irods/test/data_obj_test.py

```diff
@@ -1860,6 +1860,41 @@ def test_access_through_resc_hierarchy__243(self):
         finally:
             s.resources.remove('parent')
 
+
+    def test_temporary_xml_mode_change_with_operation_as_proof__issue_586(self):
+        from irods.message import (XML_Parser_Type, current_XML_parser, string_for_XML_parser)
+        active_xml_for_thread = lambda : string_for_XML_parser(current_XML_parser())
```

Comment:
Consider moving lines 1865 and 1866 to where they are first used. That keeps the reader from needing to "remember" so many things.

Reply:
Done.  Tried to do the same for the other test of #586 as well, but interdependencies were a little constricting, so left it alone.